src/components/search-bar/SearchBar.tsx

```diff
@@ -26,12 +37,12 @@ function SearchBar({ pageTitle, count, clearFilters }: SearchBarProps) {
     setIsListVisible(!isListVisible);
   };
 
-  const selectClients = timeEntries.map((timeEntry) => timeEntry.client).sort();
+  const sortedClientNames = timeEntries.map((timeEntry) => timeEntry.client).sort();
 
-  const clientSet = new Set(selectClients);
-  const clientArray = Array.from(clientSet);
+  const clientSet = new Set(sortedClientNames);
+  const clientNames = Array.from(clientSet);
 
-  const clientList = clientArray.map((client, index) => {
+  const clientList = clientNames.map((client, index) => {
```

Comment:
You're currently showing the client filter options based on the visible time entries, I don't think that's what you want. These filter options should probably always stay the same, no matter which time entries are visible at that moment. Maybe create an additional useState (`filteredTimeEntries` or something) and use those time entries when a filter is active, so you always still have the original set of time entries in `timeEntries`.